packages/remirror__extension-link/src/link-extension.ts

```diff
@@ -739,6 +792,55 @@ export class LinkExtension extends MarkExtension<LinkOptions> {
 
     return autoLinkAllowedTLDs.includes(tld);
   }
+
+  private findURLInString(input: string): { url: string; startIndex: number } | undefined {
+    const urlMatch = findMatches(input, this.options.autoLinkRegex)[0];
+
+    if (!urlMatch) {
+      return;
+    }
+
+    const url = input.slice(urlMatch.index, this.getURLEndIndex(input, getMatchString(urlMatch)));
```

Comment:
Slice adjacent characters that would result in an invalid URL.

Using "getURLEndIndex" we are able to determine the end of a valid URL which is slightly more ambiguous than determining the start of a URL. 

For the start, we can simply use the "urlMatch.index"